components/dialog/dialog-confirm.js

```diff
@@ -0,0 +1,58 @@
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { DialogMixin } from './dialog-mixin.js';
+import { dialogStyles } from './dialog-styles.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { heading3Styles } from '../typography/styles.js';
+
+class DialogConfirm extends DialogMixin(LitElement) {
+
+	static get properties() {
+		return {
+			text: { type: String }
+		};
+	}
+
+	static get styles() {
+		return [ dialogStyles, heading3Styles, css`
+			.d2l-dialog-outer {
+				max-width: 420px;
+			}
+			.d2l-dialog-content > div {
+				padding-top: 30px;
+			}
+			.d2l-dialog-header + .d2l-dialog-content > div {
+				padding-top: 0;
+			}
+		`];
+	}
+
+	render() {
+		if (!this._titleId) this._titleId = getUniqueId();
+		if (!this._textId) this._textId = getUniqueId();
+
+		const inner = html`
+			<div class="d2l-dialog-inner">
+				${this.title ? html`
+					<div class="d2l-dialog-header">
+						<div><h2 id="${this._titleId}" class="d2l-heading-3">${this.title}</h2></div>
+					</div>` : null}
+				<div id="${this._textId}" class="d2l-dialog-content">
+					<div>${this.text}</div>
```

Comment:
Similarly, is this wrapper `<div>` needed for something? CSS is getting applied to it, but could the padding not have gone on `.d2l-dialog-content`?

Reply:
Originally I had just the one `div` but I think I was running into some measurement problems  (timing-wise) on the flex element. It was pretty frustrating, I can give it another try though.

Reply:
So I revisited this and I think I have an explanation.  First, here is the behavior:

![dialog-resize](https://user-images.githubusercontent.com/9042472/62810159-18e06f80-bacb-11e9-991e-42951b53baee.gif)

The vertical layout is:
```
dialog (flex, direction vertical)
   header (flex none)
   content (flex)
   footer (flex none)
```
The height of the "content" is handled automatically by the browser when we set the total height of the dialog.  The height we apply to the dialog is calculated by adding the scroll-heights of header, content, and footer.  As we resize the viewport to be narrower, the scroll-height of the content increases, and we increase the height of the dialog if possible.  No problem... works as expected.  

However, the trouble arises when we subsequently resize to make the viewport wide again. 
 We try to recalculate the new height but dialog height is fixed at it's previous value still, resulting in the content element's scroll-height also not changing (because it's height is being controlled by the browser per our flex rule).  And so, as a result, the dialog doesn't shrink, leaving whitespace below.  By measuring a div inside, we can get the real scroll-height of the content inside (unaffected by the flex layout), and in turn update the dialog height with the value we expect.